// @simplewebauthn/server/dist/helpers/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifySignature = exports.validateCertificatePath = exports.toHash = exports.parseAuthenticatorData = exports.isCertRevoked = exports.isBase64URLString = exports.getCertificateInfo = exports.generateChallenge = exports.decodeCredentialPublicKey = exports.decodeClientDataJSON = exports.decodeCborFirst = exports.decodeAttestationObject = exports.convertPublicKeyToPEM = exports.convertCOSEtoPKCS = exports.convertCertBufferToPEM = exports.convertAAGUIDToString = void 0;
const convertAAGUIDToString_1 = __importDefault(require("./convertAAGUIDToString"));
exports.convertAAGUIDToString = convertAAGUIDToString_1.default;
const convertCertBufferToPEM_1 = __importDefault(require("./convertCertBufferToPEM"));
exports.convertCertBufferToPEM = convertCertBufferToPEM_1.default;
const convertCOSEtoPKCS_1 = __importDefault(require("./convertCOSEtoPKCS"));
exports.convertCOSEtoPKCS = convertCOSEtoPKCS_1.default;
const convertPublicKeyToPEM_1 = __importDefault(require("./convertPublicKeyToPEM"));
exports.convertPublicKeyToPEM = convertPublicKeyToPEM_1.default;
const decodeAttestationObject_1 = __importDefault(require("./decodeAttestationObject"));
exports.decodeAttestationObject = decodeAttestationObject_1.default;
const decodeCbor_1 = require("./decodeCbor");
Object.defineProperty(exports, "decodeCborFirst", { enumerable: true, get: function () { return decodeCbor_1.decodeCborFirst; } });
const decodeClientDataJSON_1 = __importDefault(require("./decodeClientDataJSON"));
exports.decodeClientDataJSON = decodeClientDataJSON_1.default;
const decodeCredentialPublicKey_1 = __importDefault(require("./decodeCredentialPublicKey"));
exports.decodeCredentialPublicKey = decodeCredentialPublicKey_1.default;
const generateChallenge_1 = __importDefault(require("./generateChallenge"));
exports.generateChallenge = generateChallenge_1.default;
const getCertificateInfo_1 = __importDefault(require("./getCertificateInfo"));
exports.getCertificateInfo = getCertificateInfo_1.default;
const isBase64URLString_1 = __importDefault(require("./isBase64URLString"));
exports.isBase64URLString = isBase64URLString_1.default;
const isCertRevoked_1 = __importDefault(require("./isCertRevoked"));
exports.isCertRevoked = isCertRevoked_1.default;
const parseAuthenticatorData_1 = __importDefault(require("./parseAuthenticatorData"));
exports.parseAuthenticatorData = parseAuthenticatorData_1.default;
const toHash_1 = __importDefault(require("./toHash"));
exports.toHash = toHash_1.default;
const validateCertificatePath_1 = __importDefault(require("./validateCertificatePath"));
exports.validateCertificatePath = validateCertificatePath_1.default;
const verifySignature_1 = __importDefault(require("./verifySignature"));
exports.verifySignature = verifySignature_1.default;
