// @simplewebauthn/server/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingsService = exports.MetadataService = exports.verifyAuthenticationResponse = exports.generateAuthenticationOptions = exports.verifyRegistrationResponse = exports.generateRegistrationOptions = void 0;
/**
 * @packageDocumentation
 * @module @simplewebauthn/server
 */
const generateRegistrationOptions_1 = __importDefault(require("./registration/generateRegistrationOptions"));
exports.generateRegistrationOptions = generateRegistrationOptions_1.default;
const verifyRegistrationResponse_1 = __importDefault(require("./registration/verifyRegistrationResponse"));
exports.verifyRegistrationResponse = verifyRegistrationResponse_1.default;
const generateAuthenticationOptions_1 = __importDefault(require("./authentication/generateAuthenticationOptions"));
exports.generateAuthenticationOptions = generateAuthenticationOptions_1.default;
const verifyAuthenticationResponse_1 = __importDefault(require("./authentication/verifyAuthenticationResponse"));
exports.verifyAuthenticationResponse = verifyAuthenticationResponse_1.default;
const metadataService_1 = __importDefault(require("./services/metadataService"));
exports.MetadataService = metadataService_1.default;
const settingsService_1 = __importDefault(require("./services/settingsService"));
exports.SettingsService = settingsService_1.default;
